lib/src/main/scala/spinal/lib/formal/FormalMasterSlave.scala

```diff
@@ -0,0 +1,18 @@
+package spinal.lib.formal
+
+import spinal.core._
+import spinal.lib.IMasterSlave
+
+trait FormalBundle {
+  self =>
+  type Self <: FormalBundle
+
+  def formalIsValid() : Bool
+  def formalAssertEquivalence(that : Self): Unit = {}
+}
+
+trait FormalMasterSlave extends IMasterSlave with FormalBundle {
```

Comment:
If you want the FormalMasterSlave to be a traits which is required to be used with IMaterSlave, then is that possible to use Self-Type Annotation? So that it is not necessary to change the design related code.
``` Scala
trait FormalMasterSlave extends FormalBundle { self: IMasterSlave =>
  ......
}
```

Moreover, what is the purpose of  `formallsValid` and `formalAssertEquivalence` methods? It's a bit hard to me.